### ISSUE TITLE
Email is no longer created when notification is triggered

### ISSUE THREAD

**Issue by dinoperovic** [awesto/django-shop#554]
I realised that after upgrading to 0.10.x the Email object is no longer created when the notification is triggered.

It seems the `models.nofification.order_event_notification` fails here:
```python
if not isinstance(instance, OrderModel):
    return
```

Debugging with `raise ValueError(instance, OrderModel)` before this line outputs:
```bash
ValueError: (<Order(pk=3)>, <MaterializedModel: <class 'shopit.models.order.Order'>>)
```

Replacing `OrderModel` with `BaseOrder` when checking instance fixes the issue here, but I'm not sure it's right. It didn't change with 0.10.0 release as far as I can tell.

**Comment by rfleschenberg** [comment]
I can confirm this, I ran into the same bug.

**Comment by rfleschenberg** [comment]
Suggest fix in #555, but needs review.
